.github/workflows/e2e_upgrade.yml

```diff
@@ -0,0 +1,156 @@
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: checkout upstream
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '3'
+          path: ./upstream
+          repository: ${{ github.event.pull_request.base.repo.full_name }}
+          ref: ${{ github.event.pull_request.base.ref }}
+
+      - name: Set up Go
+        uses: ./head/.github/actions/setup_env
+
+      - name: Build MatrixOne for head
+        run: |
+          cd $GITHUB_WORKSPACE/head && make clean && make build
+          git rev-parse --short HEAD          
+
+      - name: Build MatrixOne for upstream
+        run: |
+          cd $GITHUB_WORKSPACE/upstream && make clean && make build          
+          git rev-parse --short HEAD
+
+      - name: Create path for execution
+        run: |
+          cd $GITHUB_WORKSPACE && mkdir workspace 
+
+      - name: Start the Latest Version of Main of MO
+        run: |
+          cd $GITHUB_WORKSPACE/workspace
+          mv $GITHUB_WORKSPACE/upstream/etc .
+          mv $GITHUB_WORKSPACE/upstream/mo-service .
+          mv $GITHUB_WORKSPACE/upstream/optools .
+          echo "==============cn configuration of upstram=============="
+          cat ./etc/launch/cn.toml
+          echo "==============dn configuration of upstram=============="
+          cat ./etc/launch/dn.toml     
+          ./optools/run_bvt.sh $GITHUB_WORKSPACE/workspace launch
+
+      - name: Clone test-tool repository
+        uses: actions/checkout@v3
+        with:
+          repository: matrixorigin/mo-tester
+          path: ./mo-tester
+          ref: main
+
+      - name: Set up JDK 8 for x64
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          architecture: x64
+
+      - name: Start BVT Test for MO with the Latest Version of Main
+        id: bvt_on_latest_main_version
+        run: |
+          export LC_ALL="C.UTF-8"
+          locale
+          cd $GITHUB_WORKSPACE/mo-tester
+          ./run.sh -n -g -p $GITHUB_WORKSPACE/upstream/test/distributed/cases 2>&1
+
+      - name: Dump mo-service goroutines
+        if: ${{ always() && !cancelled() }}
+        run: |
+          max_retry=5
+          retry=0
+          while [ "$(ps -ef | grep 'mo-service' | grep -v "grep" | wc -l)" -gt 0 ]; do
+            curl http://localhost:12345/debug/pprof/goroutine\?debug=2 -o ${{ github.workspace }}/workspace/upstream-dump-stacks${retry}.log
+            pkill -9 mo-service
+            retry=$((retry+1))
+            if [ ${retry} -ge ${max_retry} ]; then
+              echo 'after retry, still cannot shutdown mo-service'
+              exit 1
+            fi
+            sleep 2
+          done
+
+      - name: Start the Version of Head of MO
+        run: |
+          cd $GITHUB_WORKSPACE/workspace
+          mv mo-service mo-service-upstream
+          mv mo-service.log upstream-mo-service.log
+          rm -rf mo-data/etl
+          rm -rf etc
```

Comment:
need add this for head
~~~bash
# delete for clear the start finish status of mo
rm -rf mo-data/local/system_init_completed
~~~